public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -127,6 +135,15 @@ function getLegend(props: Props, displayValues: FieldDisplay[]) {
   );
 }
 
+function hasFrames(fieldDisplayValues: FieldDisplay[]) {
+  for (let fd of fieldDisplayValues) {
+    if (fd.view && fd.view.dataFrame.length > 0) {
```

Comment:
```suggestion
    if (fd.view?.dataFrame.length) {
```
I think this. can be simplified a bit, altho I'd probably go with smth like 
```
function hasFrames(fieldDisplayValues: FieldDisplay[]) {
  return fieldDisplayValues.some((fd) => fd.view?.dataFrame.length);
}
```

Reply:
Yeah that would clean it up a bit, thanks.